xarray/core/dask_array_compat.py

```diff
@@ -130,6 +131,10 @@ def _validate_pad_output_shape(input_shape, pad_width, output_shape):
         )
 
 
+def is_duck_dask_array(x):
+    return is_array_like(x) and isinstance(x, dask_array_type)
```

Comment:
Yes, it should be `is_dask_collection(x)`. It would be okay for `is_dask_collection` to be defined as `None` upon an `ImportError` exception, right?

Reply:
I'd make that either `lambda _: True` or `lambda _: False` (not sure which)

Reply:
Since `x` cannot be a Dask collection if Dask is not importable, I think `is_dask_collection` should be
```python
def is_dask_collection(x):
    return False
```
when the `ImportError` occurs.

Reply:
It feels a bit awkward to put a function definition in a try/except, so I think `lambda _: False` works better.

Reply:
> It feels a bit awkward to put a function definition in a try/except, so I think `lambda _: False` works better.

I didn't think
```
is_dask_collection = lambda _: False
```
was allowed in flake8, hence the suggestion on regular function definition. But, if you can get away with it, then sure!

Reply:
Flake8 allows it.

Reply:
> Flake8 allows it.

Good to know!

Reply:
that's because we've configured `flake8` to suppress E731, otherwise `flake8` would complain